Fix #3037: Add ResetPedVoice function - Bug fix

TracerDS opened the pull request:
The function was bugged; it was never actually resetting ped's voice due to internal mta's logic.
Now should be fixed, function returns `true` when the voice was successfully reset, returns `false` if the voice was not changed (ie. player joins the server, player executes `resetPedVoice` without ever calling `setPedVoice`)

The old PR (#3042) was merged, so I had to make new PR

sbx320 commented on Client/game_sa/CPedSA.h:
```diff
@@ -265,24 +264,24 @@ class CPedSA : public virtual CPed, public virtual CPhysicalSA
-    CPedIntelligenceSA* m_pPedIntelligence;
-    CPedSAInterface*    m_pPedInterface;
-    CPedSoundSA*        m_pPedSound;
+    CPedIKSA*           m_pPedIK = NULL;
```
nullptr

tederis commented on Client/game_sa/CPedSoundSA.h:
```diff
@@ -70,15 +70,22 @@ class CPedSoundSA : public CPedSound
     CPedSoundSAInterface* m_pInterface;
 
 public:
+    CPedSoundSA(CPedSoundSAInterface&& pInterface) noexcept
```
What is the purpose of rvalue reference here?

TracerDS replied:
to distinguish it from the normal sound

tederis replied:
Nonsence! When you're using a rvalue reference you should use it as a rvalue reference. 

```
CPedSoundSA(CPedSoundSAInterface* pInterface, bool bInstantiate = false) :
        m_pInterface(bInstantiate ? new CPedSoundSAInterface() : pInterface)
{
        if (bInstantiate)
        {
                SetVoiceID(pInterface->m_sVoiceID);
                SetVoiceTypeID(pInterface->m_sVoiceType);
        }
}
```
would be much better

tederis commented on Client/game_sa/CPedSA.cpp:
```diff
@@ -94,7 +88,7 @@ void CPedSA::Init()
     m_pPedIntelligence = new CPedIntelligenceSA(m_pPedIntelligenceInterface, this);
     m_pPedSound = new CPedSoundSA(&pedInterface->pedSound);
-    m_pDefaultPedSound = new CPedSoundSA(&pedInterface->pedSound);
+    m_pDefaultPedSound = new CPedSoundSA(std::move(pedInterface->pedSound));
```
What exactly are you trying to move here? CPedSoundSA constructor below doesn't implement any move.

TracerDS replied:
`std::move` transforms pedSound into a rvalue. pedSound need to be a copy and by `std::move`ing it, it kinda achieves it but I could've better written it now that I think of it

tederis replied:
This is counterintuitive and breaks principles of the C++ standard.

TracerDS committed:
Finally fixed the bug

Im done doing all the CPedSoundSA mess.
Instead I do a bit of a trickery and manually create default voice type and id variables and set them on initialisation.

TracerDS committed:
Refactored CClientPed class

Only small refactors but it will make it easier for future refactors

TracerDS committed:
Refactored CPedSA constructor a little

TracerDS committed:
Removed noexcept from CClientPed

In order to change that, everything else should be refactored.
There are ~32 classes that inherit GetType() from CClientEntity.
Its just better to leave it for future refactoring

Dutchman101 committed:
Merge branch 'master' into 250523_Add_resetPedVoice_function_#3037
